### PR TITLE
Processing Data in Models

### PR DESCRIPTION
**Pull request by jugosack**
#### In this pull request is completed the following requirements:

**Create models**

- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Remember that author_id column in posts table should be the foreign_key for the users table.
- Use models to insert data using the rails console
- Create custom methods
- Your models should include some additional methods.

**Users**

- A method that returns the 3 most recent posts for a given user.

**Posts**

- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

**Comments**

- A method that updates the comments counter for a post.

**Likes**

- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+
+  def recent_post
+    # posts.order (:created_at :desc).limit(3)
```
Comment (Ismailco) [inline app/models/user.rb]: - [ ] Kindly make sure to remove this commented code, so the base code looks clean and easy to navigate

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+  after_save :update_posts_counter
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```
Comment (Ismailco) [inline app/models/post.rb:8]: - [ ] [OPTIONAL] You can make the ` update_posts_counter`method private

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @jugosack, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Clean code
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @jugosack, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did a great job removing the commented code, the codebase looks clean now.

## Optional suggestions

- [ ] This project should have the CSS linters set up as well, I strongly advise you to add it and make it pass before you merge because it will be easy to work in other milestones. The PR would have been invalidated. I'm sure the first reviewer missed this part.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**